stdlib/sys.mli

```diff
@@ -240,110 +241,130 @@ type signal_behavior =
 
-val set_signal : int -> signal_behavior -> unit
+val set_signal : signal -> signal_behavior -> unit
 (** Same as {!Sys.signal} but return value is ignored. *)
 
 
 (** {2 Signal numbers for the standard POSIX signals.} *)
 
-val sigabrt : int
+val sigabrt : signal
 (** Abnormal termination *)
 
-val sigalrm : int
+val sigalrm : signal
 (** Timeout *)
 
-val sigfpe : int
+val sigfpe : signal
 (** Arithmetic exception *)
 
-val sighup : int
+val sighup : signal
 (** Hangup on controlling terminal *)
 
-val sigill : int
+val sigill : signal
 (** Invalid hardware instruction *)
 
-val sigint : int
+val sigint : signal
 (** Interactive interrupt (ctrl-C) *)
 
-val sigkill : int
+val sigkill : signal
 (** Termination (cannot be ignored) *)
 
-val sigpipe : int
+val sigpipe : signal
 (** Broken pipe *)
 
-val sigquit : int
-(** Interactive termination *)
+val sigquit : signal
+(** Signaleractive termination *)
 
-val sigsegv : int
+val sigsegv : signal
 (** Invalid memory reference *)
 
-val sigterm : int
+val sigterm : signal
 (** Termination *)
 
-val sigusr1 : int
+val sigusr1 : signal
 (** Application-defined signal 1 *)
 
-val sigusr2 : int
+val sigusr2 : signal
 (** Application-defined signal 2 *)
 
-val sigchld : int
+val sigchld : signal
 (** Child process terminated *)
 
-val sigcont : int
+val sigcont : signal
 (** Continue *)
 
-val sigstop : int
+val sigstop : signal
 (** Stop *)
 
-val sigtstp : int
-(** Interactive stop *)
+val sigtstp : signal
+(** Signaleractive stop *)
 
-val sigttin : int
+val sigttin : signal
 (** Terminal read from background process *)
 
-val sigttou : int
+val sigttou : signal
 (** Terminal write from background process *)
 
-val sigvtalrm : int
+val sigvtalrm : signal
 (** Timeout in virtual time *)
 
-val sigprof : int
-(** Profiling interrupt *)
+val sigprof : signal
+(** Profiling signalerrupt *)
 
-val sigbus : int
+val sigbus : signal
 (** Bus error
     @since 4.03 *)
 
-val sigpoll : int
+val sigpoll : signal
 (** Pollable event
     @since 4.03 *)
 
-val sigsys : int
+val sigsys : signal
 (** Bad argument to routine
     @since 4.03 *)
 
-val sigtrap : int
-(** Trace/breakpoint trap
+val sigtrap : signal
+(** Trace/breakposignal trap
     @since 4.03 *)
 
-val sigurg : int
+val sigurg : signal
 (** Urgent condition on socket
     @since 4.03 *)
 
-val sigxcpu : int
+val sigxcpu : signal
 (** Timeout in cpu time
     @since 4.03 *)
 
-val sigxfsz : int
+val sigxfsz : signal
 (** File size limit exceeded
     @since 4.03 *)
 
+val sigemt : signal
+(** Emulate instruction executed
+    @since 5.4 *)
+
+val sigio : signal
+(** I/O is possible on a descriptor
+    @since 5.4 *)
+
+val sigwinch : signal
+(** Window size change
+    @since 5.4 *)
+
+val siginfo : signal
+(** Status request from keyboard
+    @since 5.4 *)
+
+val signal_to_string : signal -> string
+(** Signal name as a string, based on POSIX names.
+    Raises [Not_found] exception for unrecognised signal.
+    @since 5.4 *)
```

Comment:
What is the use-case of this function? Debugging? Also, we try to avoid exceptions for new library functions in general.

Reply:
> What is the use-case of this function? Debugging?

Reporting signals to end-users, e.g. after a [signaling](https://github.com/ocaml/ocaml/blob/f08e8a1ad48013dbdefc0e5415c2bf48a6881de8/otherlibs/unix/wait.c#L59-L62) `Unix.waitpid`. If you print the bare int, end-users won't be able to make sense of OCaml's renumbering. 

But yes raising `Not_found` is really not a good idea (especially since you don't have a function to check which signals are recognized…). Usually [that's the way](https://github.com/b0-system/b0/blob/93841fa33a9c65f07aef0b5d8982b78d25a2a8aa/src/std/b0__fmt.mli#L209-L212) I do it.

Reply:
My use case is also for reporting signals to end users, I've written that function across a few different projects and it looks like others have too. I'll remove the exception and just do a default string including the signal number.

Reply:
Please document the string format more exactly (see e.g. the doc string I linked to), knowing how the formatting looks is essential when you integrate the string in end user messages.